pkg/base/config.go

```diff
@@ -220,70 +244,83 @@ func (cfg *Config) PGURL(user *url.Userinfo) (*url.URL, error) {
 func (cfg *Config) GetCertificateManager() (*security.CertificateManager, error) {
 	cfg.certificateManager.once.Do(func() {
 		cfg.certificateManager.cm, cfg.certificateManager.err =
-			security.NewCertificateManager(os.ExpandEnv(cfg.SSLCertsDir))
+			security.NewCertificateManager(cfg.SSLCertsDir)
 	})
 	return cfg.certificateManager.cm, cfg.certificateManager.err
 }
 
+// InitializeNodeTLSConfigs tries to load client and server-side TLS configs.
+// It also enables the reload-on-SIGHUP functionality on the certificate manager.
+// This should be called early in the life of the server to make sure there are no
+// issues with TLS configs.
+func (cfg *Config) InitializeNodeTLSConfigs(stopper *stop.Stopper) error {
+	if cfg.Insecure {
+		return nil
+	}
+
+	if _, err := cfg.GetServerTLSConfig(); err != nil {
+		return err
+	}
+	if _, err := cfg.GetClientTLSConfig(); err != nil {
+		return err
+	}
+
+	cm, err := cfg.GetCertificateManager()
+	if err != nil {
+		return nil
```

Comment:
probably a good idea. I think it got stuck in my head because barring inconvenient reload, the previous calls will error out if this one does.